vlm/src/main/scala/geotrellis/contrib/vlm/spark/RasterSourceRDD.scala

```diff
@@ -83,6 +95,9 @@ object RasterSourceRDD {
 
     sourcesRDD.persist()
 
+    val layerMetadataK: TileLayerMetadata[K] =
+      layerMetadata.copy[K](bounds = KeyBounds(sourcesRDD.map(_._1).min(), sourcesRDD.map(_._1).max()))
```

Comment:
I didn't resolve a problem of a `Spatial` Metadata yet, and resolved it in a pretty bad way. For sure probably it can be improved by making `RasterSummary` object smarter, that will allow to collect necessary metadata in a single run.